source/d3d8to9_base.cpp

```diff
@@ -202,13 +204,21 @@ HRESULT STDMETHODCALLTYPE Direct3D8::CreateDevice(UINT Adapter, D3DDEVTYPE Devic
 
 	IDirect3DDevice9 *DeviceInterface = nullptr;
 
-	const HRESULT hr = ProxyInterface->CreateDevice(Adapter, DeviceType, hFocusWindow, BehaviorFlags, &PresentParams, &DeviceInterface);
+	HRESULT hr = ProxyInterface->CreateDevice(Adapter, DeviceType, hFocusWindow, BehaviorFlags, &PresentParams, &DeviceInterface);
+
+	if (hr == D3DERR_DEVICELOST && pCurrentDeviceInterface)
+	{
+		pCurrentDeviceInterface->Reset(&CurrentPresentParams);
```

Comment:
Just wondering, but shouldn't it also work to reset the current device with the present parameters of the new device? That way we could avoid the second global variable.